File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    const queue = [];
    
    for (let i = 0; i < prices.length; i++) {
        let time = 0;
        for (let j = i + 1; j < prices.length; j++) {
            time++;
            if (prices[j] < prices[i]) {
                break;
            }
        }
        queue.push(time);
    }
    return queue;
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
# 단조스택 풀이
def solution(prices):
    N = len(prices)
    answer = [0] * N
    stack = []
    
    for index in range(N):
        price = prices[index]
        while stack and prices[stack[-1]] > price:
            past_index = stack.pop()
            answer[past_index] = index - past_index
        stack.append(index)
    while stack:
        past_index = stack.pop()
        answer[past_index] = N - 1 - past_index
    return answer